core/src/main/scala/cats/data/SetOps.scala

```diff
@@ -0,0 +1,125 @@
+package cats
+package data
+
+import scala.collection.immutable.SortedSet
+
+/** A set of ops methods which should be implemented for all set like
+  * structures, both the normal and non-empty variants.
+  *
+  * The purpose of this trait is merely to help ensure consistent
+  * implementation between the different concrete types.
+  *
+  * @tparam F This is the input type we are operating on. It is the result
+  *           type for operations which can never yield an empty set.
+  * @tparam G This is the the result type for operations which ''may'' yield
+  *           an empty set.
+  *
+  * The use of the types `F` and `G` allow us to share this trait between set
+  * types which may be have an empty state, and those which many not,
+  * e.g. `OrderedSet` and `NonEmptySet`.
+  */
+private[data] trait SetOps[F[_], G[_], A] {
```

Comment:
I'd like a longer discussion on this. I'm not sure I see the need for a layer of indirection here. Why not just implement the ops directly?

Reply:
We don't have a great track record of have consistency of methods on our data classes. See: #4070. 

I was trying to avoid that pitfall again here.

I'm honestly on the fence though too, and if we were to proceed here we'd still have to have some specialized methods for `OrderedSet` and `NonEmptySet`. 

If we proceed and people don't like this, I'd be happy to just make them normal ops.